red4ext-sys/src/interop.rs

```diff
@@ -44,6 +45,54 @@ unsafe impl ExternType for CName {
     type Kind = cxx::kind::Trivial;
 }
 
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+#[repr(C)]
+pub struct TweakDBID {
+    hash: u64,
+}
+
+impl From<u64> for TweakDBID {
+    fn from(value: u64) -> Self {
+        Self { hash: value }
+    }
+}
+
+impl TweakDBID {
+    #[inline]
+    pub const fn new(str: &str) -> Self {
+        assert!(str.len() <= u8::MAX as usize);
+        let hash = crc32(str.as_bytes()).to_ne_bytes();
+        let length = str.len() as u8;
+        Self {
+            hash: u64::from_ne_bytes([hash[0], hash[1], hash[2], hash[3], length, 0u8, 0u8, 0u8]),
```

Comment:
I'm wondering if it wouldn't be easier to do it the other way around,
represent the struct as `crc: u32, len: u8, padding: [u8; 3]`

Reply:
that's what I did originally in the commits prior to [cd15696](https://github.com/jac3km4/red4ext-rs/pull/5/commits/cd15696966b6b26ebbf307fc90c134558f55fa45)